sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -53,6 +53,18 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
       tableIdent.table.toLowerCase)
   }
 
+  /** Locks for preventing driver mem waste when concurrent table instantiation */
+  private val tableCreationLocks = Striped.lazyWeakLock(100)
```

Comment:
nit: "These locks guard against multiple attempts to instantiate a table, which wastes memory."